app-misc/kmonad/kmonad-9999.ebuild

```diff
@@ -0,0 +1,30 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# ebuild generated by hackport 0.7.2.2
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour"
+inherit haskell-cabal git-r3
+
+DESCRIPTION="Advanced keyboard remapping utility"
+HOMEPAGE="https://github.com/kmonad/kmonad"
+EGIT_REPO_URI="https://github.com/kmonad/kmonad"
+
+LICENSE="MIT"
+SLOT="0/${PV}"
+KEYWORDS=""
+
+RDEPEND=">=dev-haskell/cereal-0.5.8.1:=[profile?] <dev-haskell/cereal-0.5.9:=[profile?]
+	>=dev-haskell/lens-4.17.1:=[profile?] <dev-haskell/lens-6:=[profile?]
+	>=dev-haskell/megaparsec-7.0.5:=[profile?] <dev-haskell/megaparsec-10:=[profile?]
+	>=dev-haskell/optparse-applicative-0.14.3.0:=[profile?] <dev-haskell/optparse-applicative-0.17:=[profile?]
+	>=dev-haskell/resourcet-1.2.2:=[profile?] <dev-haskell/resourcet-2:=[profile?]
+	>=dev-haskell/rio-0.1.14.0:=[profile?] <dev-haskell/rio-0.2:=[profile?]
+	>=dev-haskell/unliftio-0.2.12:=[profile?] <dev-haskell/unliftio-0.3:=[profile?]
+	>=dev-lang/ghc-8.6.3:=
+"
```

Comment:
While I don't have a problem with you adding a live ebuild, it has to be mentioned that you can easily change the dependencies in the cabal file with something like this:

```bash
CABAL_CHDEPS=(
    "lens >=4.17.1 && <4.18" "lens >=4.17.1 && <6"
    "megaparsec >=7.0.5 && <7.1" "megaparsec >=7.0.5 && <10"
    "optparse-applicative >=0.14.3.0 && <0.15" "optparse-applicative >=0.14.3.0 && <0.17"
)
```

See:
- [`CABAL_CHDEPS`](https://github.com/gentoo-haskell/gentoo-haskell/blob/master/eclass/haskell-cabal.eclass#L171-L181)
- [`cabal_chdeps`](https://github.com/gentoo-haskell/gentoo-haskell/blob/master/eclass/haskell-cabal.eclass#L760-L783)

I think you should also add the latest version from hackage, using `CABAL_CHDEPS` as appropriate.

Reply:
That is amazing! I did not know it was possible.

I tried this approach and was unable to compile kmonad because of the incompatibility with current version of GHC. I've added upper dependency bound for ghc for hackage release (which was 2 years ago in the meantime it seems that quite a bit have changed with kmonad).